Counterexamples/NotBoundedSMulMetricSpace.lean

```diff
@@ -0,0 +1,92 @@
+/-
+Copyright (c) 2025 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+import Mathlib.Analysis.InnerProductSpace.Basic
+import Mathlib.Data.Real.StarOrdered
+import Mathlib.Topology.MetricSpace.PiNat
+
+/-! # Metric spaces are not necessarily induced by a norm.
+
+In a normed space, the distance must be translation invariant and
+homogeneous (scaled by scalar multiplication).
+
+There are examples of metric spaces where distances do not satisfy these properties,
+and thus cannot even have `BoundedSMul` instances.
+
+-/
+
+namespace Counterexample
+
+section NormInduced
+
+variable {𝕜 E : Type*} [NormedAddCommGroup E] [NormedField 𝕜] [NormedSpace 𝕜 E]
+
+/-- A distance induced by a normed space must be translation invariant. -/
+lemma dist_translation_invariant (x y c : E) :
+    dist (x + c) (y + c) = dist x y := by
+  simp
+
+/-- A distance induced by a normed space must be homogeneous, which means that it is scaled
+by scalar multiplication. -/
+lemma dist_homogeneous (c : 𝕜) (x : E) :
+    dist (c • x) 0 = ‖c‖ * dist x 0 := by
+  simp [norm_smul]
+
+end NormInduced
+
+open PiCountable
+
+noncomputable local instance : MetricSpace (ℕ → ℕ) := PiNat.metricSpace
+noncomputable local instance : MetricSpace (ℕ → ℝ) := PiCountable.metricSpace
+
+@[simp]
+lemma PiCountable.dist_translation_invariant (x y c : ℕ → ℝ) :
+    dist (x + c) (y + c) = dist x y := by
+  simp [dist_eq_tsum]
+
+noncomputable instance : NormedAddCommGroup (ℕ → ℝ) where
+  __ := PiCountable.metricSpace
+  norm x := dist x 0
+  dist_eq x y := by
+    simpa [← sub_eq_add_neg] using
+      (PiCountable.dist_translation_invariant x y (-y)).symm
+
+lemma PiCountable.norm_single (i : ℕ) (r : ℝ) :
```

Comment:
I think this and the instance above should be in mathlib proper, since the metric space instance is?

Reply:
The metric space instance is a def. Do you want me to do the `local instance` in the `PiNat` file?

Reply:
Yes, exactly.

Reply:
`Normed*` isn't imported in the `PiNat` file. Would you like a separate file made?

Reply:
Yes, a new file sounds good.